docs/generators.md

```diff
@@ -0,0 +1,53 @@
+# Generators
+
+**Generators** are classes that are used to render models for a given language.
+
+## Supported languages
+
+- [JavaScript](../src/generators/javascript/JavaScriptGenerator.ts),
+- [TypeScript](../src/generators/typescript/TypeScriptGenerator.ts),
+- [Java](../src/generators/java/JavaGenerator.ts).
+
+## Generator's options
+
+Each generator extends default options for the generator. It means that the generator can also have additional options.
+
+Options is passed as first argument to generator's constructor. Check the example:
+
+```ts
+const generator = new TypeScriptGenerator({ ...options });
+```
+
+Default options contains:
+
+| Option | Type | Description | Default value |
+|---|---|---|---|
+| `indentation` | Object | Options for indentation. | - |
+| `indentation.type` | String | Type of indentation. Its value can be either `SPACES` or `TABS`. | `SPACES` |
+| `indentation.size` | String | Size of indentation. | 2 |
+| `defaultPreset` | Object | Default preset for generator. For more information, read [customisation](./customisation.md) document. | _Implemented by generator_ |
+| `presets` | Array | Array contains **presets**. For more information, read [customisation](./customisation.md) document. | `[]` |
```

Comment:
```suggestion
| `presets` | Array | Array contains **presets**. For more information, read [customization](./customisation.md) document. | `[]` |
```

Reply:
Like above.

Reply:
the file is called `customisation.md` with `s`. Meaning we use the British form. Do we have any agreement on what to use? (US VS British English forms)

Reply:
No, we don't (as I know) - I used `s` because grammarly highlighted me the errors for `z`.

Reply:
your grammarly is just set for British English, I recommend switching to US English as it is just a standard in technical writing.

for me, this is not a blocker for this PR because we are in general not consistent in this matter. I would just, for now, recommend always follow US English as this is standard worldwide

Reply:
Ok, I changed to `customiZation` everywhere :) Thanks!